pennylane/ops/functions/equal.py

```diff
@@ -807,3 +807,15 @@ def _equal_hilbert_schmidt(
         return False
 
     return True
+
+
+@_equal_dispatch.register
+def _equal_prep_sel_prep(
+    op1: PrepSelPrep, op2: PrepSelPrep, **kwargs
+):  # pylint: disable=unused-argument
+    """Determine whether two PrepSelPrep are equal"""
+    if op1.wires != op2.wires:
+        return f"op1 and op2 have different wires. Got {op1.wires} and {op2.wires}."
+    if not qml.equal(op1.lcu, op2.lcu):
+        return f"op1 and op2 have different lcu. Got {op1.lcu} and {op2.lcu}"
```

Comment:
Maybe we should also check `if op1.control != op2.control`? Otherwise two PrepSelPreps with the same wire sets, but different controls could be considered equal.